// colors-chalk-package/index.js
// index.js

const chalk = require('chalk');

// Function to format the current date and time
function getFormattedDate() {
  const now = new Date();

  const options = {
    month: '2-digit',
    day: '2-digit',
    year: 'numeric',
    hour: 'numeric',
    minute: '2-digit',
    second: '2-digit',
    hour12: true,
  };

  return now.toLocaleString('en-US', options);
}

// Function to simulate delay and log
function delayedLog(delay = 2000) {
  const start = Date.now();

  setTimeout(() => {
    const pid = process.pid;
    const dateTime = getFormattedDate();
    const end = Date.now();
    const elapsedTime = `${end - start}ms`;

    // Define colors
    const pidColor = chalk.green("[Nest] " + pid.toString() + "  -");
    const logColor = chalk.hex('#98c379')('LOG');
    const nestAppColor = chalk.yellowBright('[NestApplication]');
    const messageColor = chalk.hex('#98c379')('Nest application successfully started');
    const msColor = chalk.yellowBright(`+${elapsedTime}`);

    // Construct the log message
    const logMessage = `${pidColor} ${dateTime}     ${logColor} ${nestAppColor} ${messageColor} ${msColor}`;

    console.log(logMessage);
  }, delay);
}

// Export the delayedLog function
module.exports = delayedLog;

// If the script is run directly, execute the function
if (require.main === module) {
  delayedLog();
}
